test/src/test/java/org/corfudb/runtime/collections/PersistedCorfuTableTest.java

```diff
@@ -0,0 +1,1006 @@
+            intended.forEach(value -> table.insert(value, value));
+            Assertions.assertEquals(table.size(), intended.size());
+            intended.forEach(value -> Assertions.assertEquals(table.get(value), value));
+            intended.forEach(table::delete);
+
+            final Set<String> persisted = table.entryStream().map(Map.Entry::getValue).collect(Collectors.toSet());
+            Assertions.assertTrue(persisted.isEmpty());
+        }
+    }
+
+    @Property(tries = NUM_OF_TRIES)
+    void txInsertGetRemove(@ForAll @Size(SAMPLE_SIZE) Set<String> intended) {
+        resetTests();
+        try (final PersistedCorfuTable<String, String> table = setupTable()) {
+            executeTx(() -> intended.forEach(value -> table.insert(value, value)));
+            Assertions.assertEquals(table.size(), intended.size());
+            executeTx(() -> {
+                intended.forEach(value -> Assertions.assertEquals(table.get(value), value));
+                intended.forEach(table::delete);
+            });
+
+            executeTx(() -> {
+                final Set<String> persisted = table.entryStream().map(Map.Entry::getValue).collect(Collectors.toSet());
+                Assertions.assertTrue(persisted.isEmpty());
+            });
+        }
+    }
+
+    /**
+     * Verify basic non-transactional get and insert operations.
+     */
+    @Property(tries = NUM_OF_TRIES)
+    void nonTxGetAndInsert(@ForAll @Size(SAMPLE_SIZE) Set<String> intended) {
+        resetTests();
+        try (final PersistedCorfuTable<String, String> table = setupTable()) {
+            intended.forEach(value -> table.insert(value, value));
+            intended.forEach(k -> Assertions.assertEquals(k, table.get(k)));
+        }
+    }
+
+    /**
+     * Verify commit address of transactions for disk-backed tables.
+     */
+    @Property(tries = NUM_OF_TRIES)
+    void verifyCommitAddressMultiTable(@ForAll @Size(SAMPLE_SIZE) Set<String> intended) {
+        resetTests();
+        try (final PersistedCorfuTable<String, String> table1 = setupTable();
+             final PersistedCorfuTable<String, String> table2 = setupTable(alternateTableName)) {
+            table1.insert(defaultNewMapEntry, defaultNewMapEntry);
+
+            assertThat(executeTx(() -> {
+                table1.get(nonExistingKey);
+                table2.get(nonExistingKey);
+            })).isZero();
+
+            intended.forEach(value -> table2.insert(value, value));
+            assertThat(executeTx(() -> {
+                table1.get(nonExistingKey);
+                table2.get(nonExistingKey);
+            })).isEqualTo(intended.size());
+        }
+    }
+
+    /**
+     * Verify commit address of interleaving transactions on disk-backed tables.
+     */
+    // TODO(ZacH):
+    @Property(tries = NUM_OF_TRIES)
+    void verifyCommitAddressInterleavingTxn(@ForAll @Size(SAMPLE_SIZE) Set<String> intended) throws Exception {
```

Comment:
When setting ENABLE_READ_YOUR_WRITES to false, this test still has assertions failing. The commit address being returned from T1 is 0 as opposed to 100, while verifyCommitAddressMultiTable crashes.

Reply:
Seems like the issue is here, as `version` is obtained from `baseSnapshotVersion`

```
            return getAndCacheSnapshotProxy(proxy, ts)
                    .access(accessFunction, version -> updateKnownStreamPosition(proxy, version));
```

Reply:
Fixed and added new tests.